System information v2

luisgg98 opened the pull request:
I've added actuator to the project, it provides the information we committed on the project contract.
I followed these tutorials:
https://www.baeldung.com/spring-boot-actuators
https://www.baeldung.com/spring-boot-actuators#boot-2x-actuator

Commit:
First information
Not sure if it's the work information, I have to check it

Commit:
I was doing everything wrong
In the last committed I wasn't really sure if the values that I was receiving were correct,
therefore I decided to visualize the memory consume and memory avaible on the terminal and see
if it matches with the values that I had. It didn't so I have repeated if

Commit:
Tests added

Commit:
Cleaning code

Commit:
Little bug corrected

Commit:
Metrics and database information available at /info/metrics or /info/database
// TODO: Testing

Commit:
Tests on endpoints working

Commit:
Tests on endpoints working

AlbertoCalvoRubio commented on src/main/resources/application.properties:
```diff
@@ -3,4 +3,32 @@ spring.datasource.platform=hsqldb
+
+
+
+#endpoints.health.sensitive=true
+#management.security.enabled=false
+# There is no security
```
Is this necessary?

luisgg98 replied:
No, I'll erase it

AlbertoCalvoRubio commented on src/main/java/urlshortener/service/ShortURLService.java:
```diff
@@ -38,4 +38,8 @@ public ShortURL save(String url, String sponsor, String ip) {
+
+  public Long getTotalURL() {
+    return shortURLRepository.count();
+  } // NOSONAR
```
Why is it necessary here? What do you want to surpress?

luisgg98 replied:
![image](https://user-images.githubusercontent.com/45603226/99911653-35231280-2cf6-11eb-96ee-300f5f1506b1.png)
Sorry I thought the whole function was covered by the comment

AlbertoCalvoRubio reviewed (changes requested):
Check the comments and refactor that lines if it is necessary

Commit:
Comments solved

Commit:
Reorganized

luisgg98 committed:
Merge branch 'master' into system-information

Commit:
Restructured
Lombok added
Qr improved
Code http changed
System Information reorganized
Frontend updated

Commit:
Resolved errors

AlbertoCalvoRubio commented on src/main/java/urlshortener/domain/UrlStatus.java:
```diff
@@ -4,7 +4,8 @@
   INVALID("Invalid url"),
-  UNREACHABLE("Unreachable url");
+  UNREACHABLE("Unreachable url"),
+  IDNOTFOUND("ID of shortened URL not found");
```
This isn't a URL Status. It makes no sense to put this here.

luisgg98 replied:
So I'll create a local variable instead

AlbertoCalvoRubio commented on src/main/java/urlshortener/service/ClickService.java:
```diff
@@ -25,4 +24,8 @@ public void saveClick(String hash, String ip) {
             : "[" + hash + "] was not saved");
   }
+
+  public Long getTotalClick() { // NOSONAR
+    return clickRepository.count(); // NOSONAR
```
Why //NOSONAR?

luisgg98 replied:
![image](https://user-images.githubusercontent.com/45603226/100548135-0e6f5980-326b-11eb-97c6-8b70ddc6e15c.png)
Same mistake again sorry

AlbertoCalvoRubio commented on src/test/java/urlshortener/web/UrlShortenerTests.java:
```diff
@@ -132,6 +133,7 @@ public void thatShortenerFailsIfTheURLisNotSafe() throws Exception {
         .andExpect(status().isBadRequest());
   }
 
+  //ok
```
?

luisgg98 replied:
I was checking one by one if the tests were working I didn't realize I left that comment

AlbertoCalvoRubio reviewed (changes requested):
Read comments below and answer the questions

Commit:
Comments removed
